### PR TITLE
(avo-2604): Share change role modal

### PR DESCRIPTION
**Pull request by SHDBramVandenbussche**
[AVO-2604](https://meemoo.atlassian.net/browse/AVO-2604)

image

[AVO-2604]: https://meemoo.atlassian.net/browse/AVO-2604?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/assignment/components/AssignmentActions.tsx [hunk 1]
```diff
@@ -71,7 +70,12 @@ const AssignmentActions: FunctionComponent<AssignmentActionsProps> = ({
 				config?.button?.className
 			)}
 		>
-			<ShareAssignmentWithPupil {...share} {...config} />
+			<ShareDropdown
+				users={mockShareUsers}
+				onAddNewUser={(value: Partial<ShareUserInfo>) => console.log(value)}
+				onDeleteUser={(value) => console.log(value)}
+				onEditRights={(user, newRights) => console.log(user, newRights)}
+			/>
```
Comment (WardVerc1) [inline src/assignment/components/AssignmentActions.tsx:78]: mockdata and console.log's will be replaced with actual data, right?

File: src/shared/components/ShareDropdown/ShareDropdown.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	padding-top: 0 !important;
+}
+
+.c-dropdown__content-open {
+	border: $border;
+	border-radius: 0.5rem;
+	background-color: $g-base-background;
+	margin-top: 1rem;
+	width: 545px;
+	filter: drop-shadow(0px 4px 4px rgba($color-navy, 0.25));
```
Comment (WardVerc1) [inline src/shared/components/ShareDropdown/ShareDropdown.scss:15]: use rem

File: src/shared/components/ShareWithColleagues/ShareWithColleagues.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	display: grid;
+	row-gap: 1rem;
+}
+
+.c-colleague-info-row {
+	display: grid;
+	align-items: center;
+	column-gap: 1rem;
+	grid-template-columns: max-content 1fr 1fr 7%;
+	font-size: 15px;
```
Comment (WardVerc1) [inline src/shared/components/ShareWithColleagues/ShareWithColleagues.scss:73]: Dont we have a variable for font-size? If not, use rem

File: src/shared/components/ShareWithColleagues/ShareWithColleagues.scss [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	&__avatar {
+		.c-avatar img {
+			width: 100%;
+			height: 100%;
+		}
+	}
+
+	&__info {
+		&__email {
+			font-size: 14px;
```
Comment (WardVerc1) [inline src/shared/components/ShareWithColleagues/ShareWithColleagues.scss:85]: rem

File: src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+	const [error, setError] = useState<string | null>(null);
+	const [isEditRightsModalOpen, setIsEditRightsModalOpen] = useState<boolean>(false);
+	const [toEditShareUser, setToEditShareUser] = useState<ShareUserInfo>();
+
+	const handleRightsButtonClicked = () => {
+		setIsRightsDropdownOpen(!isRightsDropdownOpen);
+	};
+
+	const handleAddNewUser = () => {
+		if (/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(newShareUser.email as string)) {
```
Comment (WardVerc1) [inline src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx:53]: Extract this regex to a const with a descriptive name so it's more clear what it does

File: src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+										separator: ' ',
+									})}
+								</p>
+							</div>
+
+							<div className="c-colleague-info-row__rights">
+								<span>{shareUserRightToString(user.rights)}</span>
+
+								{currentUser.rights === ShareUserInfoRights.OWNER &&
+									currentUser.email !== user.email && (
```
Comment (WardVerc1) [inline src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx:123]: Extract this condition to a const with a descriptive name so it's more clear what it checks
ex. "isOwner"

**Review by WardVerc1 — CHANGES_REQUESTED** [review]

File: src/assignment/components/AssignmentActions.tsx (hunk 1 above)
Comment (SHDBramVandenbussche) [reply src/assignment/components/AssignmentActions.tsx:78]: Yes

File: src/shared/components/ShareDropdown/ShareDropdown.scss [hunk 7]
```diff
@@ -11,8 +11,8 @@ $border: 1px solid $color-gray-200;
 	border-radius: 0.5rem;
 	background-color: $g-base-background;
 	margin-top: 1rem;
-	width: 545px;
-	filter: drop-shadow(0px 4px 4px rgba($color-navy, 0.25));
+	width: 54rem;
+	filter: drop-shadow(0rem 0.25rem 0.25rem rgba($color-navy, 0.25));
```
Comment (bertyhell) [inline src/shared/components/ShareDropdown/ShareDropdown.scss:15]:
```suggestion
	filter: drop-shadow(0 0.25rem 0.25rem rgba($color-navy, 0.25));
```

File: src/shared/components/ShareWithColleagues/Modals/EditShareUserRightsModal.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+	currentRight: ShareUserInfoRights;
+};
+
+const EditShareUserRightsModal: FC<EditShareUserRightsModalProps> = ({
+	isOpen,
+	handleClose,
+	handleConfirm,
+	currentRight,
+}) => {
+	const [right, setRight] = useState<ShareUserInfoRights>();
```
Comment (bertyhell) [inline src/shared/components/ShareWithColleagues/Modals/EditShareUserRightsModal.tsx:33]:
```suggestion
	const [right, setRight] = useState<ShareUserInfoRights>(currentRight);
```

File: src/shared/components/ShareWithColleagues/Modals/EditShareUserRightsModal.tsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+	isOpen: boolean;
+	handleClose: () => void;
+	handleConfirm: (right: ShareUserInfoRights) => void;
+	currentRight: ShareUserInfoRights;
+};
+
+const EditShareUserRightsModal: FC<EditShareUserRightsModalProps> = ({
+	isOpen,
+	handleClose,
+	handleConfirm,
```
Comment (bertyhell) [inline src/shared/components/ShareWithColleagues/Modals/EditShareUserRightsModal.tsx:30]: Personal preference: 
for component parameters i usually use: onClose, onConfirm
for internal methods that need to handle some work before calling the onClose/onConfirm i use: handleClose, handleConfirm

But that is just personal preference.

**Review by bertyhell — APPROVED** [review]

**Review by WardVerc1 — APPROVED** [review]

**Comment by WardVerc1** [comment]
When clicking on the dropdown, the other flyout is closed: 
image
image
image


Also after changing from 'Bijdrager' to 'Kijker' , the value is not changed, the user remains 'Bijdrager'

**Review by WardVerc1 — CHANGES_REQUESTED** [review]
See comment

**Pull request merged**